UploadServiceLogo/handler.ts

```diff
@@ -0,0 +1,141 @@
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorTooManyRequests,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+
+import {
+  checkSourceIpForHandler,
+  clientIPAndCidrTuple as ipTuple
+} from "io-functions-commons/dist/src/utils/source_ip_check";
+
+import { Context } from "@azure/functions";
+import { identity } from "fp-ts/lib/function";
+import {
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import { Logo } from "../generated/api-admin/Logo";
+import { APIClient } from "../utils/clients/admin";
+import {
+  ErrorResponses,
+  IResponseErrorUnauthorized,
+  toErrorServerResponse
+} from "../utils/responses";
+
+type ResponseTypes =
+  | IResponseSuccessJson<undefined>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+/**
+ * Type of a GetUploadServiceLogoHandler handler.
+ *
+ * GetUploadServiceLogo expects a service ID and a logo as input
+ * and returns informations about upload outcome
+ */
+type IGetUploadServiceLogoHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  serviceId: string,
+  logoPayload: Logo
+) => Promise<ResponseTypes>;
+
+const uploadServiceLogoTask = (
+  apiClient: ReturnType<APIClient>,
+  serviceId: string,
+  logo: Logo
+): TaskEither<ErrorResponses, IResponseSuccessJson<undefined>> =>
+  tryCatch(
+    () =>
+      apiClient.uploadServiceLogo({
+        logo,
+        service_id: serviceId
+      }),
+    errs => ResponseErrorInternal(JSON.stringify(errs))
+  ).foldTaskEither(
+    err => fromLeft(err),
+    maybeResponse =>
+      maybeResponse.fold(
+        errs => fromLeft(ResponseErrorInternal(JSON.stringify(errs))),
+        responseType =>
+          responseType.status !== 201
+            ? fromLeft(toErrorServerResponse(responseType))
+            : taskEither.of(ResponseSuccessJson(responseType.value))
+      )
+  );
+
+/**
+ * Handles requests for upload a service logo by a service ID and a base64 logo' s string.
+ */
+export function GetUploadServiceLogoHandler(
+  apiClient: ReturnType<APIClient>
+): IGetUploadServiceLogoHandler {
+  return (_, __, ___, ____, serviceId, logoPayload) => {
+    return uploadServiceLogoTask(apiClient, serviceId, logoPayload)
```

Comment:
must add the same check as in updateService

Reply:
Fixed in [98db5c7](https://github.com/pagopa/io-functions-services/pull/68/commits/98db5c77eefe2f2bbce99559596e00feed83aa0c)